src/services/telementryService.js

```diff
@@ -159,11 +169,20 @@ return (process.env.REACT_APP_TELEMETRY_MODE === 'ET' && currentMode === 'ET') |
 
 export const getEventOptions = () => {
   var emis_username = 'anonymous';
+  var buddyUserId = '';
+
   if (localStorage.getItem('token') !== null) {
     let jwtToken = localStorage.getItem('token');
     var userDetails = jwt(jwtToken);
     emis_username = userDetails.emis_username;
   }
+
+  if (localStorage.getItem('buddyToken') !== null) {
```

Comment:
instead of checking the localStorage you can check the same this same variable "isBuddyLogin"